litert/vendors/qualcomm/core/transformation/mha_to_sha.cc

```diff
@@ -786,12 +777,296 @@ size_t OptimizeMHAFastVlmPrefill(
               ops.begin() + start_index + pattern_size);
     return step_size;
   }
+  QNN_LOG_WARNING(
+      "[G2G] Validation failed. Rolling back to the original graph.");
```

Comment:
NIT: 
More information in the warning to locate the issue.
```suggestion
      "[G2G] Validation failed in TransformToSHA. Rolling back to the original graph.");
```